### PR TITLE
Cleanup of bi_flush()

### PR DESCRIPTION
**Pull request by Dead2**
In CI and my personal testing, we never hit this `if (s->bi_valid == 64)` condition.
That does not mean that it never hits, but I propose that it might not even be worth having the code there.
To still be able to promise <7 bits left (instead of 8), add case handling 48 bytes.

Also I noticed that `bi_flush()` is only used two places, and one of those functions is only calling `bi_flush()`. Therefore rename to `zng_tr_flush_bits()` to simplify things.

### CODE REVIEW

**Comment by Dead2** [comment]
## Develop 20498320cdd38ac6db265596598e47e2d0602850 Feb 7
```
 Level   Comp   Comptime min/avg/max/stddev  Decomptime min/avg/max/stddev  Compressed size
 1     54.166%      0.150/0.154/0.155/0.001        0.033/0.036/0.038/0.001        8,523,732
 2     43.871%      0.243/0.248/0.249/0.001        0.030/0.035/0.038/0.002        6,903,609
 3     42.387%      0.291/0.295/0.298/0.002        0.031/0.035/0.036/0.001        6,670,099
 4     41.647%      0.319/0.324/0.326/0.002        0.030/0.034/0.036/0.001        6,553,723
 5     41.216%      0.345/0.350/0.353/0.002        0.031/0.034/0.035/0.001        6,485,938
 6     41.037%      0.389/0.397/0.401/0.003        0.029/0.032/0.035/0.001        6,457,776
 7     40.778%      0.496/0.499/0.503/0.002        0.028/0.033/0.035/0.002        6,416,919
 8     40.704%      0.607/0.613/0.619/0.004        0.028/0.033/0.035/0.001        6,405,244
 9     40.409%      0.908/0.918/0.921/0.003        0.030/0.033/0.035/0.001        6,358,951

 avg1  42.913%                        0.422                          0.034
 tot                                113.921                          9.168       60,775,991

   text    data     bss     dec     hex filename
 138278    1352       8  139638   22176 libz-ng.so.2
```
## PR #1660
```
 Level   Comp   Comptime min/avg/max/stddev  Decomptime min/avg/max/stddev  Compressed size
 1     54.166%      0.151/0.153/0.155/0.001        0.033/0.035/0.037/0.001        8,523,732
 2     43.871%      0.241/0.248/0.250/0.002        0.032/0.036/0.038/0.002        6,903,609
 3     42.387%      0.293/0.296/0.298/0.001        0.031/0.034/0.036/0.002        6,670,099
 4     41.647%      0.319/0.323/0.326/0.002        0.031/0.034/0.035/0.001        6,553,723
 5     41.216%      0.345/0.350/0.353/0.002        0.032/0.034/0.035/0.001        6,485,938
 6     41.037%      0.391/0.396/0.399/0.002        0.030/0.033/0.035/0.001        6,457,776
 7     40.778%      0.493/0.499/0.503/0.003        0.030/0.033/0.035/0.001        6,416,919
 8     40.704%      0.603/0.612/0.617/0.004        0.030/0.033/0.035/0.001        6,405,244
 9     40.409%      0.911/0.916/0.921/0.003        0.030/0.033/0.035/0.001        6,358,951

 avg1  42.913%                        0.422                          0.034
 tot                                113.807                          9.180       60,775,991

   text    data     bss     dec     hex filename
 138190    1352       8  139550   2211e libz-ng.so.2
 ```
Benchmarks are within margins of error I'd say, and I would not expect a significant change since this is not used in any hot paths.

File: trees.c [hunk 1]
```diff
@@ -790,27 +782,26 @@ static int detect_data_type(deflate_state *s) {
...
-            s->bi_buf >>= 8;
-            s->bi_valid -= 8;
-        }
+void Z_INTERNAL zng_tr_flush_bits(deflate_state *s) {
+    if (s->bi_valid >= 48) {
+        put_uint32(s, (uint32_t)s->bi_buf);
+        put_short(s, (uint16_t)(s->bi_buf >> 32));
+        s->bi_buf >>= 48;
+        s->bi_valid -= 48;
+    }else if (s->bi_valid >= 32) {
```
Comment (nmoinvaz) [inline trees.c]: Missing space.

Comment (Dead2) [reply trees.c]: Fixed

**Review by nmoinvaz — APPROVED** [review]

**Review by nmoinvaz — APPROVED** [review]
Nice idea! LGTM.

**Comment by mtl1979** [comment]
The code is only reachable when using `Z_PARTIAL_FLUSH`,  but I don't see any CI tests using it....

**Pull request merged**